### PR TITLE
Wip improvements solution

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
@jmburges

### CODE REVIEW

File: app/models/song.rb [hunk 1]
```diff
@@ -1,6 +1,45 @@
 class Song
   # code here
 
+  attr_accessor :name, :artist
+  attr_reader :genre
+
+  @@all = []
+
+  def initialize
+    self.class.all << self
```
Comment (jmburges) [inline app/models/song.rb]: why are you doing this?

File: spec/07_library_parser_spec.rb [hunk 2]
```diff
@@ -13,19 +13,19 @@
...
   it 'builds a song based on song parts' do
-    parts = ['Action Bronson', 'Larry Csonka', 'indie']
-    song = parser.build_song(parts[0], parts[1], parts[2])
+    # parts = ['Action Bronson', 'Larry Csonka', 'indie']
+    song = parser.build_song('db/data/Action Bronson - Larry Csonka [indie].mp3')
 
-    expect(Artist.find_by_name(parts[0])).to eq(song.artist)
-    expect(Song.find_by_name(parts[1])).to eq(song)
-    expect(Genre.find_by_name(parts[2])).to eq(song.genre)
+    expect(Artist.find_by_name("Action Bronson")).to eq(song.artist)
```
Comment (jmburges) [inline spec/07_library_parser_spec.rb]: hmmm this is testing two things? It is testing both the `find_by_name` method as well as the build_song stuff.

Why aren't you just doing `expect(song.artist).to eq("Action Bronson")`?

Comment (SophieDeBenedetto) [inline spec/07_library_parser_spec.rb]: fixed.

File: app/models/song.rb (hunk 1 above)
Comment (SophieDeBenedetto) [inline app/models/song.rb]: fixed.

**Comment by SophieDeBenedetto** [comment]
@jmburges addressed all comments with latest commit.

**Pull request closed without merging**